src/pow_fpga_accel.c

```diff
@@ -0,0 +1,162 @@
+/* 
+ * Copyright (C) 2018 dcurl Developers. 
+ * Copyright (c) 2018 Ievgen Korokyi. 
+ * Use of this source code is governed by MIT license that can be 
+ * found in the LICENSE file.
+ */
+
+#include <fcntl.h>
+#include <sys/mman.h>
+#include <unistd.h>
+#include "trinary.h"
+#include "constants.h"
+#include "pow_fpga_accel.h"
+
+#define HPS_TO_FPGA_BASE 0xC0000000
+#define HPS_TO_FPGA_SPAN 0x0020000
+#define HASH_CNT_REG_OFFSET 4
+#define TICK_CNT_LOW_REG_OFFSET 5
+#define TICK_CNT_HI_REG_OFFSET 6
+#define MWM_MASK_REG_OFFSET 3
+#define CPOW_BASE 0
+
+/* Set FPGA configuration for device files */
+#define CTRL_FPGA_POW "/dev/cpow-ctrl"
+#define IDATA_FPGA_POW "/dev/cpow-idata"
+#define ODATA_FPGA_POW "/dev/cpow-odata"
+
+static FILE *ctrl_fd;
+static FILE *in_fd;
+static FILE *out_fd;
+static int devmem_fd;
+static void *fpga_regs_map;
+static uint32_t *cpow_map;
+
+int pow_fpga_accel_init()
+{
+    ctrl_fd = 0;
+    in_fd = 0;
+    out_fd = 0;
+    devmem_fd = 0;
+    fpga_regs_map = 0;
+    cpow_map = 0;
+
+    ctrl_fd = fopen(CTRL_FPGA_POW, "r+");
+
+    if (ctrl_fd == NULL) {
+        perror("cpow-ctrl open fail");
+        return 0;
+    }
+
+    in_fd = fopen(IDATA_FPGA_POW, "wb");
+
+    if (in_fd == NULL) {
+        perror("cpow-idata open fail");
+        fclose(ctrl_fd);
+        return 0;
+    }
+
+    out_fd = fopen(ODATA_FPGA_POW, "rb");
+
+    if (out_fd == NULL) {
+        perror("cpow-odata open fail");
+        fclose(ctrl_fd);
+        fclose(in_fd);
+        return 0;
+    }
+
+    devmem_fd = open("/dev/mem", O_RDWR | O_SYNC);
```

Comment:
The "/dev/mem" should also be used as macro.

Reply:
I think this device driver don't be changed by develops to access to the system's physical memory. Therefore, I prefer to not use macro.

Reply:
okay.